### PR TITLE
feat(spices): add border style

### PR DESCRIPTION
**Pull request by ironmanvim**
add border style utils to create borders

### CODE REVIEW

File: spices/spices.mli [hunk 1]
```diff
@@ -31,6 +32,7 @@ val reverse : bool -> style -> style
 val strikethrough : bool -> style -> style
 val underline : bool -> style -> style
 val width : int option -> style -> style
+val border : Border.t -> style -> style
```
Comment (leostera) [inline spices/spices.mli:35]: **suggestion**. from looking at how we're using this call (`border (get_border Thick)`) I think we could just export predefined borders from the Border module and use them like this, without needing a `get_border` function:

```ocaml
let overlay_border fmt =
  Spices.(default |> border Border.double |> build) fmt
```

Comment (leostera) [inline spices/spices.mli:35]: **fix**. let's add a signature for the Border module here, like:

```ocaml
module Border : sig
  type t

  val make : 
    ?bottom:string ->
    ?bottom_left:string ->
    ?bottom_right:string ->
    ?left:string ->
    ?middle:string ->
    ?middle_bottom:string ->
    ?middle_left:string ->
    ?middle_right:string ->
    ?middle_top:string ->
    ?right:string ->
    ?top:string ->
    ?top_left:string ->
    ?top_right:string ->
    unit ->
    t

  val normal : t
  val rounded : t
  (* other predefined borders here *)
end
```

**Review by leostera — CHANGES_REQUESTED** [review]
Hi @ironmanvim! Thanks for the PR ✨ this looks great. Just some minor tweaks I think we want to have here and we should be good to merge :)

Tweaks:

* adding an explicit signature to the Border module with a `make` constructor function

* exposing predefined borders directly as values rather than through a function – so we don't need a `get_border` function

* add the border example to the `examples/README.md` file

How does this sound? Let me know if you have any questions 🙏🏼

**Comment by ironmanvim** [comment]
@leostera Worked with the changes.

File: spices/border.ml [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  middle_left : string option;
+  middle_right : string option;
+  middle : string option;
+  middle_top : string option;
+  middle_bottom : string option;
+}
+
+let make ?top ?left ?bottom ?right ?top_left ?top_right ?bottom_left
+    ?bottom_right ?middle_left ?middle_right ?middle ?middle_top ?middle_bottom
+    (_unit : unit) =
```
Comment (leostera) [inline spices/border.ml:42]:
```suggestion
    () =
```

**Review by leostera — APPROVED** [review]
LGTM! 🚀

**Comment by leostera** [comment]
Thanks a lot @ironmanvim! Super happy with this :)

**Pull request merged**